optuna/terminator/improvement/evaluator.py

```diff
@@ -128,3 +128,53 @@ def _validate_input(
                 "The intersection search space is empty. This condition is not supported by "
                 f"{cls.__name__}."
             )
+
+
+@experimental_class("3.2.0")
```

Comment:
A nit, but the next release will be `3.4.0`.
```suggestion
@experimental_class("3.4.0")
```